superset/assets/javascripts/explorev2/components/SelectField.jsx

```diff
@@ -26,6 +26,40 @@ const defaultProps = {
 };
 
 export default class SelectField extends React.Component {
+  constructor(props) {
+    super(props);
+    const options = this.props.choices.map((c) => {
```

Comment:
what do you think of pulling  L31-58 into another function called `getOptions` to keep that logic out of the constructor?